madanalysis/layout/plotflow_for_dataset.py

```diff
@@ -84,58 +88,133 @@ def CreateHistogram(self):
     def ComputeScale(self):
 
         iplot = 0
+
+        with open(
+            os.path.join(self.main.archi_info.ma5dir, "madanalysis/input/LHAPDF.txt"), "r"
+        ) as f:
+            pdf_list = [int(line.split(",")[0]) for line in f.readlines()[1:]]
+
         # Loop over plot
-        for iabshisto, select in enumerate(self.main.selection):
+        for iabshisto in range(0, len(self.main.selection)):
 
             # Keep only histogram
-            if select.__class__.__name__ != "Histogram":
+            if self.main.selection[iabshisto].__class__.__name__ != "Histogram":
                 continue
 
-            processor = HistogramProcessor(
-                self.histos[iplot],
-                self.dataset.weight_collection,
-                self.dataset.measured_global.nevents,
-                self.xsection,
-            )
-
             # Reset scale
             scale = 0.0
-
-            # integral
-            integral = (
-                self.histos[iplot].positive.integral
-                - self.histos[iplot].negative.integral
-            )
-
-            integral = np.mean(integral)
+            multiweight_scale = 0.0
+            if self.multiweight_histos[iplot]:
+                self.multiweight_histos[iplot].set_central_weight_loc(
+                    scale_choice=self.dataset.dynamic_scale_choice,
+                    n_point_scale_variation=self.dataset.n_point_scale_variation,
+                    central_pdfs=pdf_list,
+                )
 
             # Case 1: Normalization to ONE
-            if select.stack == StackingMethodType.NORMALIZE2ONE or (
+            if self.main.selection[
+                iabshisto
+            ].stack == StackingMethodType.NORMALIZE2ONE or (
                 self.main.stack == StackingMethodType.NORMALIZE2ONE
                 and self.main.selection[iabshisto].stack == StackingMethodType.AUTO
             ):
+                integral = (
+                    self.histos[iplot].positive.integral
+                    - self.histos[iplot].negative.integral
+                )
+                multiweight_integral = 0
+                if self.multiweight_histos[iplot]:
+                    multiweight_integral = self.multiweight_histos[iplot].integral
                 if integral > 0.0:
                     scale = 1.0 / integral
                 else:
                     scale = 0.0
+                if multiweight_integral > 0.0:
+                    multiweight_scale = 1.0 / multiweight_integral
```

Comment:
I think this is not what should be done We should instead to normalise wrt the central scale so that only the central curve is normalised to 1 (as in typical scale variation plots).

Reply:
I tried to do exactly that (if there is no mistake that is)

https://github.com/MadAnalysis/madanalysis5/blob/98c6d1cd77b2079035af30b8a9267ae44c7f2103/madanalysis/layout/plotflow_for_dataset.py#L127

Takes the integral with respect to the central curve:

https://github.com/MadAnalysis/madanalysis5/blob/98c6d1cd77b2079035af30b8a9267ae44c7f2103/madanalysis/multiweight/histogram.py#L402-L410

Reply:
lets please keep them both until we validate everything